### ISSUE TITLE
Over-lenient string validator

### ISSUE THREAD

**Issue by msteckyefantis** [vandium-io/lov#2]
**Lov** currently allows an object to be validated using the string validator.

``` .js
const lov = require( 'lov' );

let not_a_string = {key:'value'};

let stringSchema = lov.string().required();

let result = lov.validate( not_a_string, stringSchema );

//result.error is null
//result.value is '[object Object]'
```

Using **Joi**, the same validation gives an error:

``` .js
const Joi = require( 'Joi' );

let not_a_string = {key:'value'};

let stringSchema = Joi.string().required();

let result = Joi.validate( not_a_string, stringSchema );

/*
result.error is:
 { [ValidationError: "value" must be a string]
  isJoi: true,
  name: 'ValidationError',
  details: 
   [ { message: '"value" must be a string',
       path: 'value',
       type: 'string.base',
       context: [Object] } ],
  _object: { key: 'value' },
  annotate: [Function] }


result.value is: { key: 'value' }
typeof result.value === 'object'
*/
```

---

Similar for numbers:

``` .js
const lov = require( 'lov' );

let not_a_string = 22;

let stringSchema = lov.string().required();

let result = lov.validate( not_a_string, stringSchema );

//result.error is null
//result.value is '22'
```

and **Joi:**

``` .js
const Joi = require( 'Joi' );

let not_a_string = 22;

let stringSchema = Joi.string().required();

let result = Joi.validate( not_a_string, stringSchema );

/*
result.error is:
{ [ValidationError: "value" must be a string]
  isJoi: true,
  name: 'ValidationError',
  details: 
   [ { message: '"value" must be a string',
       path: 'value',
       type: 'string.base',
       context: [Object] } ],
  _object: 22,
  annotate: [Function] }


result.value is: 22 
typeof result.value === 'number'
*/
```

---

I think making the validation more strict could improve Lov.

**note**: [Joi makes value conversion an option](https://github.com/hapijs/joi/blob/v9.0.0-2/API.md#validatevalue-schema-options-callback)

**Comment by msteckyefantis** [comment]
Also:

validating an empty string `''` with

`lov.string().trim().min( 1 ).max( 42 )`

converts the empty string to:
`'[object Object]'`

**Comment by richardhyatt** [comment]
Resolved in #0d8fa95

**Comment by richardhyatt** [comment]
Fixed in  0d8fa95
